src/components/ceremony/ceremony.service.ts

```diff
@@ -56,36 +51,44 @@ export class CeremonyService {
       throw new InputException('No ceremony id to search for', 'ceremony.id');
     }
 
-    const dto = await this.ceremonyRepo.readOne(id, session);
-    return await this.secure(dto, session);
+    const dto = await this.repo.readOne(id, session);
+    return this.secure(dto, session);
   }
 
   async readMany(ids: readonly ID[], session: Session) {
-    const ceremonies = await this.ceremonyRepo.readMany(ids, session);
-    return await Promise.all(
-      ceremonies.map((dto) => this.secure(dto, session)),
-    );
+    const ceremonies = await this.repo.readMany(ids, session);
+    return ceremonies.map((dto) => this.secure(dto, session));
   }
 
-  async secure(dto: UnsecuredDto<Ceremony>, session: Session) {
+  secure(dto: UnsecuredDto<Ceremony>, session: Session) {
     return this.privileges.for(session, Ceremony).secure(dto);
   }
 
-  async update(input: UpdateCeremony, session: Session): Promise<Ceremony> {
-    const object = await this.readOne(input.id, session);
-    const changes = this.ceremonyRepo.getActualChanges(object, input);
+  async update(
+    input: UpdateCeremony,
+    session: Session,
+  ): Promise<DbTypeOf<Ceremony>> {
+    const object = await this.repo.readOne(input.id, session);
+    const changes = this.repo.getActualChanges(object, input);
     this.privileges.for(session, Ceremony, object).verifyChanges(changes);
-    return await this.ceremonyRepo.update(object, changes);
+    const updated = await this.repo.update(
+      {
+        id: input.id,
+        ...changes,
+      },
+      session,
+    );
+    return updated;
```

Comment:
The return type of this update method should still be `Promise<Ceremony>`.
You just need to wrap it in a secure()
```suggestion
    return this.secure(updated, session);
```